packages/collector/test/tracing/database/ioredis/test.js

```diff
@@ -24,596 +24,633 @@ const globalAgent = require('../../../globalAgent');
 const mochaSuiteFn =
   semver.gte(process.versions.node, '14.0.0') && supportedVersion(process.versions.node) ? describe : describe.skip;
 
-mochaSuiteFn('tracing/ioredis', function () {
-  this.timeout(config.getTestTimeout());
-
-  globalAgent.setUpCleanUpHooks();
-  const agentControls = globalAgent.instance;
-  let controls;
+function checkConnection(span, setupType) {
+  if (setupType === 'cluster') {
+    expect(span.data.redis.connection).to.equal(process.env.AZURE_REDIS_CLUSTER);
+  } else {
+    expect(span.data.redis.connection).to.equal(process.env.REDIS);
+  }
+}
+
+// Please run this command on the root folder to start the single redis instance (default):
+// node bin/start-test-containers.js --redis
+//
+// Please set the environment variables to run the tests against azure redis cluster:
+//    export AZURE_REDIS_CLUSTER=team-nodejs-redis-cluster-tekton.redis.cache.windows.net:6380
+//    export AZURE_REDIS_CLUSTER_PWD=
+['default', 'cluster'].forEach(setupType => {
```

Comment:
The test kinda remained the same. Only:

- wrapped the tests to run all tests with a cluster
- added the check connection check
- added span length assertion, which discovered a bug (explained on the instrumentation)